src/box/lua/config/applier/lua.lua

```diff
@@ -0,0 +1,37 @@
+local alloc = require('internal.alloc')
+local log = require('internal.config.utils.log')
+
+local is_lua_memory_applied = false
+
+local function apply(config)
+    local configdata = config._configdata
+    local memory = configdata:get('lua.memory', {use_default = true})
+    local old_memory = alloc.getmemory()
+
+    -- If memory limit is applied for the first time or it is
+    -- increased.
+    if not is_lua_memory_applied or memory > old_memory then
+        -- The call throws an error if the user tries to limit the
+        -- memory with values < 256MB or with an amount less than
+        -- currently allocated.
+        alloc.setmemory(memory)
+
+        log.verbose(('lua.apply: set memory limit to %d'):format(memory))
+    -- If memory limit is decreased.
+    elseif memory < old_memory then
+        local name = 'lua_memory_decreased'
+        local warning = 'lua.apply: lua.memory decrease will not be applied ' ..
+                        'until the instance is restarted'
+
+        config._aboard:set({type = 'warn', message = warning}, {key = name})
```

Comment:
I'm OK, but if the underlying API allows to decrease it to some extent, can we allow to do so from the config? In the similar way: if it is impossible, report an alert and wait for reloading.

I'm OK with the refusal to descrease it in runtime, just curious.

@georgiy-belyanin @Buristan What do you think?